### PR TITLE
[fud2] op graph generation and testing

### PR DESCRIPTION
**Pull request by jku20**
This PR implements a method for randomly constructing a simple type of graph composed of layers of nodes connected by ops. It adds correctness tests based on these graphs as new test cases.

### CODE REVIEW

File: fud2/fud-core/Cargo.toml [hunk 1]
```diff
@@ -24,4 +24,5 @@ once_cell = "1.19.0"
 ariadne = "0.4.1"
 itertools.workspace = true
 rand = "0.8.5"
+rand_chacha = "0.3.1"
```
Comment (sampsyo) [inline fud2/fud-core/Cargo.toml:27]: Possibly silly question, but any particular need for a specific PRNG? Surely we don't need cryptographic PRNGs for this?

**Review by sampsyo — COMMENTED** [review]
Nice; looks awesome so far! This will be interesting for measuring performance as well…

File: fud2/fud-core/Cargo.toml (hunk 1 above)
Comment (jku20) [reply fud2/fud-core/Cargo.toml:27]: The specific prng doesn't matter, but I want the test to always reproduce (i.e. an update to `rand`s prng which changes their implementation shouldn't break the tests).

**Comment by jku20** [comment]
This doesn't do all that needs to be done related to generating graphs, and testing, but it is certainly still something. I think it's probably worth merging in this state instead of it just growing stale here though. It is certainly it's own atomic addition at this point.

This tests correctness against single, simple type of graph, a graph composed of connected layers of states. As it only tests correctness it makes no claims on completeness, e.g. a planner which always returns "no path found" will pass these tests. 

This PR also does no bench marking (which I guess was part of the point of this pr, oops).

**Review by sampsyo — APPROVED** [review]
It's looking super cool! I agree that there is surely a lot more that could be done with this (e.g., we could somehow generate graphs that have a specific path we're looking for) to test completeness, and we could make a benchmark out of this to compare performance side-by-side. But this is already in a useful state and a good foundation for those next steps.

**Pull request merged**